src/Compilers/CSharp/Portable/BoundTree/DeconstructionLocalPendingInference.cs

```diff
@@ -10,10 +10,11 @@ internal partial class DeconstructionLocalPendingInference
     {
         public BoundLocal SetInferredType(TypeSymbol type, bool success)
         {
-            Debug.Assert(type != null);
-            var syntaxNode = (SingleVariableDesignationSyntax)this.Syntax;
-            Binder.DeclareLocalVariable((SourceLocalSymbol)this.LocalSymbol, syntaxNode.Identifier, type);
-            return new BoundLocal(syntaxNode, this.LocalSymbol, constantValueOpt: null, type: type, hasErrors: this.HasErrors || !success);
+            Debug.Assert((object)type != null);
+            Debug.Assert(this.Syntax is SingleVariableDesignationSyntax);
```

Comment:
Yes. I will change it.

---
In reply to: [75711632](https://github.com/dotnet/roslyn/pull/13274#discussion_r75711632) [](ancestors = 75711632)